src/Simplex/Chat/Library/Subscriber.hs

```diff
@@ -2064,15 +2094,50 @@ processAgentMessageConn vr user@User {userId} corrId agentConnId agentMessage =
             probeMatchingMemberContact m' connectedIncognito
         else messageError "x.grp.link.mem error: invalid group link host profile update"
 
-    xGrpLinkAcpt :: GroupInfo -> GroupMember -> GroupMemberRole -> CM ()
-    xGrpLinkAcpt gInfo@GroupInfo {membership} m role = do
-      membership' <- withStore' $ \db -> do
-        updateGroupMemberStatus db userId m GSMemConnected
-        updateGroupMemberAccepted db user membership role
-      let m' = m {memberStatus = GSMemConnected}
-      toView $ CRUserJoinedGroup user gInfo {membership = membership'} m'
-      let connectedIncognito = memberIncognito membership
-      probeMatchingMemberContact m' connectedIncognito
+    xGrpLinkAcpt :: GroupInfo -> GroupMember -> GroupMemberRole -> Maybe MemberId -> CM ()
+    xGrpLinkAcpt gInfo@GroupInfo {groupId, membership} m role memberId_ = case memberId_ of
+      Nothing -> processUserAccepted -- backwards compatibility
```

Comment:
maybe we don't need it

Reply:
then members with previous version wouldn't be able to join groups. alternatively we could add XGrpLinkAdmit as separate event from moderators to host which does make sense (see api comment), and keep this as it was without optional member id